index.rst

```diff
@@ -1,61 +1,190 @@
 
-     Caption text.
+The identity management system for the Rubin Science Platform is responsible for authenticating users and maintaining their group memberships (see SQR-039_ and SQR-044_).
+A user of the Butler service will obtain a token from the identity management system and present it as part of an API call.
+The Butler service can, in turn, use that token to obtain the identity of the user and their group membership using the token API (see SQR-049_).
+With that information, it must somehow decide whether an operation is permitted.
 
-   Run: ``make html`` and ``open _build/html/index.html`` to preview your work.
-   See the README at https://github.com/lsst-sqre/lsst-technote-bootstrap or
-   this repo's README for more info.
+.. _SQR-039: https://sqr-039.lsst.io/
+.. _SQR-044: https://sqr-044.lsst.io/
+.. _SQR-049: https://sqr-049.lsst.io/
 
-   Feel free to delete this instructional comment.
+Proposed solution
+=================
 
-:tocdepth: 1
+Assume that Butler collection paths are of the form:
 
-.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+- ``/u/{username}/...``
+- ``/g/{group}/...``
+- ``/other/...``
 
-.. sectnum::
+with arbitrary structure below those top-level paths.
+The exact naming may be different.
+The important semantics are that there are two top-level path patterns, one for user-owned hierarchies and one for group-owned hierarchies, and then a third type of path that doesn't match either of those patterns.
+
+This proposal assumes that paths that do not begin with ``/u`` or ``/g`` will be written outside of the Butler service (via direct access to the Butler registry, for example) and will otherwise be public and read-only.
+
+Each collection may optionally be associated with an :abbr:`ACL (Access Control List)`.
+An ACL contains a list of groups that have access to that collection.
+
+When the Butler receives a user request, it first checks whether the request is to one of the public collections.
+
+1. If the path does not begin with ``/u`` or ``/g``, read access is allowed and write access is denied.
+
+If no access control decision is yet made, the Butler service then uses the associated token to obtain metadata about the user by making a call to the ``/auth/api/v1/user-info`` endpoint (see SQR-049_).
+This will return information such as:
+
+.. code-block:: json
+
+   {
+     "username": "alice",
+     "name": "Alice Example",
+     "uid": 124187,
+     "groups": [
+       {
+         "id": 124187,
+         "name": "alice",
+       },
+       {
+         "id": 204173,
+         "name": "example-group"
+       },
+       {
+         "id": 205671,
+         "name": "other-group"
+       }
+     ]
+   }
+
+The Butler service may (and generally should) cache this information for a given token.
+The cache should have a maximum lifetime of an hour.
```

Comment:
Maybe explain why 1 hour is appropriate and the tradeoffs of longer or shorter?

Reply:
I reconciled this with LSE-279 (making it 30 minutes) and added a reference.